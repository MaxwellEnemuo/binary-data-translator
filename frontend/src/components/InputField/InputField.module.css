.InputField {
    width: 250px;
    padding: 10px 20px;
    font-size: 12px;
    border-radius: 5px;
    border: 1px solid gray;
  }