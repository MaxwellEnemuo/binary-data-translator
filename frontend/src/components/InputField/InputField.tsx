import React from "react";
import { InputFieldProps } from "../../types";
import styles from "./InputField.module.css";

export const InputField: React.FC<InputFieldProps> = ({
  type,
  id,
  value,
  onChange,
}) => {
  return (
    <input
      className={styles.InputField}
      type={type}
      id={id}
      value={value}
      required
      onChange={onChange}
    />
  );
};
