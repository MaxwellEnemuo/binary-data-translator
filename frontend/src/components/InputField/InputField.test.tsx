import React from "react";
import { render } from "@testing-library/react";
import { InputField } from "./InputField";

describe("InputField component", () => {
  it("renders the input field correctly", () => {
    const { asFragment } = render(
      <InputField type="text" id="username" value="test" onChange={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
