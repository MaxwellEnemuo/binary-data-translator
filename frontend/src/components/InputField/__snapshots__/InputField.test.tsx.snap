// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component renders the input field correctly 1`] = `
<DocumentFragment>
  <input
    class="InputField"
    id="username"
    required=""
    type="text"
    value="test"
  />
</DocumentFragment>
`;
